packages/twenty-server/src/workspace/workspace-sync-metadata/comparators/workspace-field.comparator.ts

```diff
@@ -0,0 +1,168 @@
+          return datum.name;
+        },
+      },
+    );
+
+    // Compare fields
+    const fieldMetadataDifference = diff(
+      originalFieldMetadataMap,
+      standardFieldMetadataMap,
+    );
+
+    for (const difference of fieldMetadataDifference) {
+      const fieldName = difference.path[0];
+      // Object shouldn't have thousands of fields, so we can use find here
+      const standardFieldMetadata = standardObjectMetadata.fields.find(
+        (field) => field.name === fieldName,
+      );
+      const originalFieldMetadata = originalObjectMetadata.fields.find(
+        (field) => field.name === fieldName,
+      );
+
+      switch (difference.type) {
+        case 'CREATE': {
+          if (!standardFieldMetadata) {
+            throw new Error(
+              `Field ${fieldName} not found in standardObjectMetadata`,
+            );
+          }
+
+          result.push({
+            action: ComparatorAction.CREATE,
+            object: {
+              ...standardFieldMetadata,
+              objectMetadataId: originalObjectMetadata.id,
+            },
+          });
+          break;
+        }
+        case 'CHANGE': {
+          if (!originalFieldMetadata) {
+            throw new Error(
+              `Field ${fieldName} not found in originalObjectMetadata`,
+            );
+          }
+
+          const id = originalFieldMetadata.id;
+          const property = difference.path[difference.path.length - 1];
+
+          // If the old value and the new value are both null, skip
+          // Database is storing null, and we can get undefined here
+          if (
+            difference.oldValue === null &&
+            (difference.value === null || difference.value === undefined)
+          ) {
+            break;
+          }
+
+          if (typeof property !== 'string') {
+            break;
+          }
+
+          if (!fieldPropertiesToUpdateMap[id]) {
+            fieldPropertiesToUpdateMap[id] = {};
+          }
+
+          // If the property is a stringified JSON, parse it
+          if (
+            (fieldPropertiesToStringify as readonly string[]).includes(property)
+          ) {
+            fieldPropertiesToUpdateMap[id][property] = JSON.parse(
+              difference.value,
+            );
+          } else {
+            fieldPropertiesToUpdateMap[id][property] = difference.value;
+          }
+          break;
+        }
+        case 'REMOVE': {
+          if (!originalFieldMetadata) {
+            throw new Error(
+              `Field ${fieldName} not found in originalObjectMetadata`,
+            );
+          }
+
+          if (difference.path.length === 1) {
+            result.push({
+              action: ComparatorAction.DELETE,
+              object: originalFieldMetadata,
+            });
+          }
+          break;
+        }
+      }
+    }
+
+    if (Object.keys(fieldPropertiesToUpdateMap).length > 0) {
+      for (const [id, fieldPropertiesToUpdate] of Object.entries(
```

Comment:
Not really, because we can have multiple `CHANGE` for a field id, so we need to group them before pushing them into the array.
We could otherwise push into the array, find the index based on the id, and update the object in the array at the index, but it's more unclear I think